### PR TITLE
Design Patterns added

### PR DESCRIPTION
**Pull request by RashakDude**
## Changes proposed

- Added design-patterns.json in backend folder
- Added subcategory as design-patterns

### CODE REVIEW

**Comment by RashakDude** [comment]
@rupali-codes @CBID2 Kindly review

File: database/backend/design-patterns.json [hunk 1]
```diff
@@ -0,0 +1,156 @@
+[
+    {
+        "name": "Factory Method",
+        "description": "Factory Method is a creational design pattern that provides an interface for creating objects in a superclass, but allows subclasses to alter the type of objects that will be created.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:4]: [Proofreading]
```suggestion
        "description": "Factory Method is a creational design pattern that provides an interface for developing objects in a superclass. It allows subclasses to alter the type of objects that will be created.",
```

File: database/backend/design-patterns.json [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        "name": "Abstract Factory",
+        "description": "Abstract Factory is a creational design pattern that lets you produce families of related objects without specifying their concrete classes.",
+        "url": "https://refactoring.guru/design-patterns/abstract-factory",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Builder",
+        "description": "Builder is a creational design pattern that lets you construct complex objects step by step. The pattern allows you to produce different types and representations of an object using the same construction code.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:18]: [Proofreading]
```suggestion
        "description": "Builder is a creational design pattern that lets you construct complex objects step by step. The pattern also allows you to produce different types and representations of an object using the same construction code.",
```

File: database/backend/design-patterns.json [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        "name": "Singleton",
+        "description": "Singleton is a creational design pattern that lets you ensure that a class has only one instance, while providing a global access point to this instance.",
+        "url": "https://refactoring.guru/design-patterns/singleton",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Adapter",
+        "description": "Adapter is a structural design pattern that allows objects with incompatible interfaces to collaborate.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:39]: [Nitpick] Hmmm..."collaborate" makes this description a bit unclear. Does this pattern let object with different interfaces connect to each other?
```suggestion
        "description": "Adapter is a structural design pattern that allows objects with incompatible interfaces to collaborate.",
```

File: database/backend/design-patterns.json [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        "url": "https://refactoring.guru/design-patterns/adapter",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Bridge",
+        "description": "Bridge is a structural design pattern that lets you split a large class or a set of closely related classes into two separate hierarchies—abstraction and implementation—which can be developed independently of each other.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:46]: [Proofreading]
```suggestion
        "description": "Bridge is a structural design pattern that lets you split a large class or a set of closely related classes into two separate hierarchies—abstraction and implementation, which can be developed independently of each other.",
```

File: database/backend/design-patterns.json [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        "url": "https://refactoring.guru/design-patterns/bridge",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Composite",
+        "description": "Composite is a structural design pattern that lets you compose objects into tree structures and then work with these structures as if they were individual objects.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:53]: [Proofreading]
```suggestion
        "description": "Composite is a structural design pattern that lets you compose objects into tree structures and then work with them as if they were individual objects.",
```

File: database/backend/design-patterns.json [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        "name": "Decorator",
+        "description": "Decorator is a structural design pattern that lets you attach new behaviors to objects by placing these objects inside special wrapper objects that contain the behaviors.",
+        "url": "https://refactoring.guru/design-patterns/decorator",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Facade",
+        "description": "Facade is a structural design pattern that provides a simplified interface to a library, a framework, or any other complex set of classes.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:67]: [Proofreading]
```suggestion
        "description": "Facade is a structural design pattern that provides a simplified interface to a library, framework, or any other complex set of classes.",
```

File: database/backend/design-patterns.json [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        "name": "Flyweight",
+        "description": "Flyweight is a structural design pattern that lets you fit more objects into the available amount of RAM by sharing common parts of state between multiple objects instead of keeping all of the data in each object.",
+        "url": "https://refactoring.guru/design-patterns/flyweight",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Proxy",
+        "description": "Proxy is a structural design pattern that lets you provide a substitute or placeholder for another object. A proxy controls access to the original object, allowing you to perform something either before or after the request gets through to the original object.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:81]: [Proofreading]
```suggestion
        "description": "Proxy is a structural design pattern that lets you provide a substitute or placeholder for another object. It also controls access to the original object, allowing you to perform something either before or after the request gets through to the original object.",
```

File: database/backend/design-patterns.json [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        "name": "Iterator",
+        "description": "Iterator is a behavioral design pattern that lets you traverse elements of a collection without exposing its underlying representation (list, stack, tree, etc.).",
+        "url": "https://refactoring.guru/design-patterns/iterator",
+        "category": "backend",
+        "subcategory": "design-patterns"
+    },
+    {
+        "name": "Mediator",
+        "description": "Mediator is a behavioral design pattern that lets you reduce chaotic dependencies between objects. The pattern restricts direct communications between the objects and forces them to collaborate only via a mediator object.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:109]: [Proofreading]
```suggestion
        "description": "Mediator is a behavioral design pattern that lets you reduce chaotic dependencies between objects. The pattern also restricts direct communications between the objects and forces them to collaborate only via a mediator object.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @RashakDude! I pointed out a few typos and proofreading errors. In the future, I highly recommend running your descriptions through Grammarly before making them into a pull request.Keep on improving! 😄

File: database/backend/design-patterns.json (hunk 8 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json (hunk 7 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json (hunk 6 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json (hunk 5 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json (hunk 4 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json (hunk 3 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: FIxed changes

File: database/backend/design-patterns.json (hunk 2 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json (hunk 1 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Fixed changes

File: database/backend/design-patterns.json [hunk 9]
```diff
@@ -1,154 +1,154 @@
 [
     {
         "name": "Factory Method",
-        "description": "Factory Method is a creational design pattern that provides an interface for creating objects in a superclass, but allows subclasses to alter the type of objects that will be created.",
+        "description": "A Factory Method is a creational design pattern that provides an interface for creating objects in a superclass while allowing subclasses to choose the type of objects created.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:4]: [Proofreading] 

```suggestion
        "description": "A factory method is a creational design pattern that provides an interface for creating objects in a superclass while allowing subclasses to choose the type of objects created.",
```

File: database/backend/design-patterns.json [hunk 10]
```diff
@@ -1,154 +1,154 @@
...
         "url": "https://refactoring.guru/design-patterns/factory-method",
         "category": "backend",
         "subcategory": "design-patterns"
     },
     {
         "name": "Abstract Factory",
-        "description": "Abstract Factory is a creational design pattern that lets you produce families of related objects without specifying their concrete classes.",
+        "description": "The Abstract Factory design pattern allows you to create families of linked items without declaring their concrete classes.",
```
Comment (CBID2) [inline database/backend/design-patterns.json:11]:
```suggestion
        "description": "The abstract factory design pattern allows you to create families of linked items without declaring their concrete classes.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @RashakDude. You're close. I just made a few more proofreading suggestions.

File: database/backend/design-patterns.json (hunk 9 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Resolved

File: database/backend/design-patterns.json (hunk 10 above)
Comment (RashakDude) [reply database/backend/design-patterns.json]: Resolved

**Review by rupali-codes — APPROVED** [review]
looks good to me

**Comment by RashakDude** [comment]
Please review @ujjawaltyagii @CBID2

**Review by CBID2 — APPROVED** [review]
LGTM

**Review by ujjawaltyagii — APPROVED** [review]
Lgtm 👍

**Comment by rupali-codes** [comment]
@CBID2 why did you remove the label 👀

**Comment by CBID2** [comment]
> @CBID2 why did you remove the label 👀

Which one?

**Comment by rupali-codes** [comment]
@RashakDude hey, could you please resolve the conflicts? so we can merge your PR

**Comment by rupali-codes** [comment]
> > @CBID2 why did you remove the label 👀
> 
> Which one?

`goal: new-link`

**Comment by CBID2** [comment]
> > > @CBID2 why did you remove the label 👀
> 
> > 
> 
> > Which one?
> 
> 
> 
> `goal: new-link` 

Because the level 1 label exemplifies that category.

**Comment by rupali-codes** [comment]
@RashakDude can you please resolve merge conflicts?

**Pull request closed without merging**